### PR TITLE
Prevent automatic conversion of field names

### PR DESCRIPTION
**Pull request by toshi1127**
This PR closes the issue of Prevent automatic conversion of field names. #696 

### What are the changes and their implications?
- Prevent automatic conversion of field names.

### Checklist

- [x] Tests added for changes
- [ ] PR submitted to [blitzjs.com](https://github.com/blitz-js/blitzjs.com) for any user facing changes

### CODE REVIEW

**Comment by toshi1127** [comment]
Should I fix the list so that it doesn't also auto-convert to plural form?

```
  it("pluralizes lists and makes single fields singular", () => {
    expect(Field.parse("userIds:string[]").toString()).toEqual(
      Field.parse("userId:string[]").toString(),
    )
  })
```

**Comment by aem** [comment]
@toshi1127 yeah that would be great! Thanks so much for tackling this

File: packages/generator/test/prisma/field.test.ts [hunk 1]
```diff
@@ -62,21 +60,21 @@ describe("Field model", () => {
...
-    expect(Field.parse("hasOne:tasks?").toString()).toMatchInlineSnapshot(`"task  Task?"`)
+    expect(Field.parse("hasOne:tasks?").toString()).toMatchInlineSnapshot(`"tasks  Task?"`)
     // list identifier should be ignored for singular relations
     expect(Field.parse("hasOne:task[]").toString()).toMatchInlineSnapshot(`"task  Task"`)
-    expect(Field.parse("hasOne:tasks[]").toString()).toMatchInlineSnapshot(`"task  Task"`)
+    expect(Field.parse("hasOne:tasks[]").toString()).toMatchInlineSnapshot(`"tasks  Task"`)
   })
 
   it("handles hasMany relations", () => {
-    expect(Field.parse("hasMany:task").toString()).toMatchInlineSnapshot(`"tasks  Task[]"`)
```
Comment (toshi1127) [inline packages/generator/test/prisma/field.test.ts:74]: Prevent field name from being converted to plural forms when creating a Field with a list type property.

File: packages/generator/test/prisma/field.test.ts [hunk 2]
```diff
@@ -6,20 +6,18 @@ describe("Field model", () => {
...
     expect(Field.parse("users:int[]").toString()).toMatchInlineSnapshot(`"users  Int[]"`)
-    expect(Field.parse("user:int[]").toString()).toMatchInlineSnapshot(`"users  Int[]"`)
-  })
-
-  it("pluralizes lists and makes single fields singular", () => {
-    expect(Field.parse("name:string").toString()).toEqual(Field.parse("names:string").toString())
-    expect(Field.parse("userIds:string[]").toString()).toEqual(
-      Field.parse("userId:string[]").toString(),
-    )
+    expect(Field.parse("user:int[]").toString()).toMatchInlineSnapshot(`"user  Int[]"`)
```
Comment (toshi1127) [inline packages/generator/test/prisma/field.test.ts:20]: Prevent field name from being converted to plural forms when creating a Field with a list type property.

File: packages/generator/test/prisma/field.test.ts [hunk 3]
```diff
@@ -62,21 +60,21 @@ describe("Field model", () => {
 
   it("handles hasOne relations", () => {
```
Comment (toshi1127) [inline packages/generator/test/prisma/field.test.ts:61]: Prevent automatic conversion of field name when creating a Field with relationships, such as `hasOne`.

**Comment by flybayer** [comment]
@aem this is ready for final review

**Pull request merged**